cmd/wallet/main.go

```diff
@@ -1,163 +1,378 @@
+		if err != nil {
+			return nil, err
+		}
+		walletCredentials = &WalletCredentials{
+			accountSeed: accountSeed,
+			pk:          pk,
+			sk:          sk,
+			address:     address,
+		}
+
+	case existingBase58AccountSeed:
+		if opts.Base58AccountSeed == "" {
+			return nil, errors.Wrap(wrongProgramArguments, "no base58 account seed was provided")
+		}
+		accountSeed, err := crypto.NewDigestFromBase58(opts.Base58AccountSeed)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to decode base58-encoded account seed")
+		}
+		pk, sk, address, err := generateOnAccountSeed(accountSeed, opts.Scheme)
+		if err != nil {
+			return nil, err
+		}
+		walletCredentials = &WalletCredentials{
+			accountSeed: accountSeed,
+			pk:          pk,
+			sk:          sk,
+			address:     address,
+		}
+
+	default:
+		showUsageAndExit()
 	}
 
-	fmt.Print("Enter seed: ")
-	seed, err := gopass.GetPasswd()
+	return walletCredentials, nil
+}
+
+func createWallet(opts Opts) error {
+	walletPath, err := getWalletPath(opts.PathToWallet)
 	if err != nil {
-		fmt.Println("Interrupt")
-		return
+		return errors.Wrap(err, "failed to handle wallet's path")
+	}
+	fmt.Println(`Available options: 
+		 1: Generate new seed phrase and wallet
+		 2: Create a wallet based on an existing seed phrase. Requires seed phrase argument to be provided
+		 3: Create a wallet based on an existing base58-encoded seed phrase. Requires seed phrase argument to be provided and seed-base58 flag marked "true"
+		 4: Create a wallet based on an existing base58-encoded account seed. Requires account seed argument in Base58 format`)
+
+	var walletCredentials *WalletCredentials
+	var choice WalletGenerationChoice
+	_, err = fmt.Scanf("%s", &choice)
+	if err != nil {
+		return errors.Wrap(err, "failed to scan the choice of how to create the wallet")
+	}
+
+	walletCredentials, err = generateWalletCredentials(choice, opts)
+	if err != nil {
+		return errors.Wrap(err, "failed to generate wallet's credentials")
+	}
+	if walletCredentials == nil {
+		return errors.New("failed to generate wallet's credentials")
+	}
+
+	fmt.Print("Enter password to encode your account seed: ")
+	pass, err := gopass.GetPasswd()
+	if err != nil {
+		return errors.Wrap(err, "failed to get the password to encode the account seed")
+	}
+
+	if len(pass) == 0 {
+		return errors.Wrap(err, "the password's length is zero")
 	}
 
-	if opts.Base58 {
-		seed, err = base58.Decode(string(seed))
+	var wlt wallet.Wallet
+	if exists(walletPath) {
+		fmt.Print("Wallet already exists on the provided path. Rewrite? Y/N: ")
+		var answer string
+		_, err := fmt.Scanf("%s", &answer)
 		if err != nil {
-			fmt.Printf("Err: %s\n", err.Error())
-			return
+			return errors.Wrap(err, "failed to get the answer on rewriting the existing wallet")
+		}
+		switch answer {
+		case "Y":
+			wlt = wallet.NewWallet()
+		case "N":
+			return nil
+		default:
+			return errors.New("unknown command")
```

Comment:
What if I'll press `enter` key multiple times? Is it an error? I don't think so. I think we should consider it like `N` (this option should be set by default).